dex/asset.go

```diff
@@ -119,3 +121,30 @@ type Asset struct {
 	SwapSizeBase uint64 `json:"swapSizeBase"`
 	SwapConf     uint32 `json:"swapConf"`
 }
+
+// Denomination is a unit and its conversion factor.
+type Denomination struct {
+	Unit             string `json:"unit"`
+	ConversionFactor uint64 `json:"conversionFactor"`
+}
+
+// UnitInfo conveys information about the units and available denominations for
+// an asset.
+type UnitInfo struct {
+	// AtomicUnit is the name associated with the asset's integral unit of
+	// measure, e.g. satoshis, atoms, gwei (for DEX purposes).
+	AtomicUnit string `json:"atomicUnit"`
+	// Conventional is the conventionally-used denomination.
+	Conventional Denomination `json:"conventional"`
+	// Alternatives lists additionally available Denominations, and can be
+	// empty.
+	Alternatives []Denomination `json:"denominations"`
+}
+
+// ConventionalString converts the quantity to conventional units, and returns
+// the formatted float string.
+func (ui *UnitInfo) ConventionalString(v uint64) string {
+	c := ui.Conventional.ConversionFactor
+	prec := int(math.Round(math.Log10(float64(c)))) // Assumes integer powers of 10
+	return strconv.FormatFloat(float64(v)/float64(c), 'f', prec, 64)
+}
```

Comment:
I like `UnitInfo` a lot.  Good solution.
I think a simple unit test for the `ConventionalString` method would be helpful to explore edges and test formatting.  Mainly I was wondering about the computation of `prec` and how that improves on different formatting like `prec=-1` or `%g`.  I'm not saying it doesn't, just that some tests to ensure certain outputs would be nice.